staging/src/k8s.io/externaljwt/apis/v1/api.proto

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// To regenerate api.pb.go run `hack/update-codegen.sh protobindings`
+syntax = "proto3";
+
+package v1;
+
+option go_package = "k8s.io/externaljwt/apis/v1";
+
+import "google/protobuf/timestamp.proto";
+
+// This service is served by a process on a local Unix Domain Socket.
+service ExternalJWTSigner {
+    // Sign takes a serialized JWT payload, and returns the serialized header and
+    // signature.  The caller can then assemble the JWT from the header, payload,
+    // and signature.
+    //
+    // The plugin MUST set a key id in the returned JWT header.
+    rpc Sign(SignJWTRequest) returns (SignJWTResponse) {}
+  
+    // FetchKeys returns the set of public keys that are trusted to sign
+    // Kubernetes service account tokens. Kube-apiserver will call this RPC:
+    //
+    // * Every time it tries to validate a JWT from the service account issuer with an unknown key ID, and
+    //
+    // * Periodically, so it can serve reasonably-up-to-date keys from the OIDC
+    //   JWKs endpoint.
+    rpc FetchKeys(FetchKeysRequest) returns (FetchKeysResponse) {}
+    
+    // Metadata is meant to be called once on startup.
+    // Enables sharing metadata with kube-apiserver (eg: the max token lifetime that signer supports)
+    rpc Metadata(MetadataRequest) returns (MetadataResponse) {}
+  }
+  
+  message SignJWTRequest {
+    // URL-safe base64 wrapped payload to be signed.
+    // Exactly as it appears in the second segment of the JWT
+    string claims = 1;
+  }
+  
+  message SignJWTResponse {
+    // header must contain only alg, kid, typ claims.
+    // typ must be “JWT”.
```

Comment:
Do we have tests for the "must" statements in 57 and 58? I see the code in https://github.com/kubernetes/kubernetes/blob/d70229fac7c2aec4e6a69e0783fa0860a90e1144/pkg/serviceaccount/externaljwt/plugin/plugin.go#L167C5-L175 but i can't find the tests. same for `alg` as well?